pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -409,41 +424,35 @@ func handleModify(ctx *downloaderContext, key string,
 		errStr := fmt.Sprintf("Name changed - not allowed %s -> %s\n",
 			config.Name, status.Name)
 		log.Error(errStr)
-		status.LastErr = errStr
-		status.LastErrTime = time.Now()
+		status.SetErrorInfo(errStr)
 		publishDownloaderStatus(ctx, status)
 		return
 	}
+
+	log.Infof("handleModify(%s) RefCount %d to %d, Expired %v for %s\n",
+		status.ImageID, status.RefCount, config.RefCount,
+		status.Expired, status.Name)
+
+	status.PendingModify = true
+	publishDownloaderStatus(ctx, status)
+
 	if config.IsContainer != status.IsContainer {
 		log.Infof("handleModify: Setting IsContainer to %t for %s",
 			config.IsContainer, status.ImageID)
 		status.IsContainer = config.IsContainer
 		publishDownloaderStatus(ctx, status)
 	}
-	log.Infof("handleModify(%s) RefCount %d to %d, Expired %v for %s\n",
-		status.ImageID, status.RefCount, config.RefCount,
-		status.Expired, status.Name)
-
 	// If RefCount from zero to non-zero then do install
 	if status.RefCount == 0 && config.RefCount != 0 {
-		status.PendingModify = true
 		log.Infof("handleModify installing %s\n", config.Name)
-		handleCreate(ctx, status.ObjType, config, key)
-		status.RefCount = config.RefCount
```

Comment:
Good catch - this broken code published two separate Status!

Reply:
True.